src/main/java/core/basesyntax/Car.java

```diff
@@ -1,5 +1,67 @@
 package core.basesyntax;
 
-public class Car {
+public class Car implements Cloneable {
+    private Engine engine;
+    private String model;
+    private int height;
+    private double length;
+    private boolean isAvailable;
 
+    public Engine getEngine() {
+        return engine;
+    }
+
+    public void setEngine(Engine engine) {
+        this.engine = engine;
+    }
+
+    public String getModel() {
+        return model;
+    }
+
+    public void setModel(String model) {
+        this.model = model;
+    }
+
+    public int getHeight() {
+        return height;
+    }
+
+    public void setHeight(int height) {
+        this.height = height;
+    }
+
+    public double getLength() {
+        return length;
+    }
+
+    public void setLength(double length) {
+        this.length = length;
+    }
+
+    public boolean isAvailable() {
+        return isAvailable;
+    }
+
+    public void setAvailable(boolean available) {
+        this.isAvailable = available;
+    }
+
+    @Override
+    public Car clone() {
+        try {
+            Car clonedCar = (Car) super.clone();
+            clonedCar.setEngine(engine.clone());
+            return clonedCar;
+        } catch (CloneNotSupportedException e) {
+            throw new RuntimeException("ERROR: error occurred in 'Car' class - clone() method", e);
+        }
+    }
+
+    @Override
+    public String toString() {
+        return "Car{" + " engine = " + engine + ", carModel=" + model
+                + ", carField3=" + height + ", carField4=" + length
+                + ", carField5=" + isAvailable + "}";
```

Comment:
field names are not informative :(

Reply:
Sorry... Now I know that it is better to write a good code from start rathar then reformat a bad one...